network/server/tchannelthrift/node/service.go

```diff
@@ -350,7 +350,8 @@ func (s *service) FetchBlocksMetadataRaw(tctx thrift.Context, req *rpc.FetchBloc
 
 	for _, fetchedMetadata := range fetchedResults {
 		blocksMetadata := s.blocksMetadataPool.Get()
-		blocksMetadata.ID = fetchedMetadata.ID.Data()
+		// LEAK(@kobolog): `fetched` is closed before return, can ID expire?
+		blocksMetadata.ID = append([]byte{}, fetchedMetadata.ID.Data()...)
```

Comment:
Also: we need to Clone the ID that is put in the metadata object here in `storage/block/NewFetchBlocksMetadataResult(id ts.ID, ...)`.